src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java

```diff
@@ -0,0 +1,118 @@
+package com.google.checkstyle.test.chapter7javadoc.rule712paragraphs;
+
+/**
+ * Some Javadoc.
+ *
+ * <p>/^ WARN/ Some Javadoc.
+ *
+ * <p>
+ */
+class InputFormattedIncorrectJavadocParagraph {
+  // violation 3 lines above '<p> tag should be placed immediately before the first word'
+
+  // violation 4 lines below '<p> tag should be placed immediately before the first word'
+  /**
+   * Some Javadoc.
+   *
+   * <p>
+   *
+   * <p>Some Javadoc.
+   *
+   * @since 8.0
+   */
+  public static final byte NUL = 0;
+
+  /**
+   * Some
+   *
+   * <p>Javadoc.
+   *
+   * <p>Some Javadoc.
+   *
+   * @see <a
+   *     href="http://www.gwtproject.org/doc/latest/DevGuideOrganizingProjects.html#DevGuideModules">
+   *     Documentation about GWT emulated source</a>
+   */
+  boolean emulated() {
+    return false;
+  }
+
+  // violation 6 lines below '<p> tag should be placed immediately before the first word'
+  // violation 7 lines below '<p> tag should be placed immediately before the first word'
+  // violation 8 lines below '<p> tag should be placed immediately before the first word'
+  /**
+   * Some Javadoc.
+   *
+   * <p>
+   *
+   * <p>
+   *
+   * <p>
+   *
+   * <p>/^WARN/ Some Javadoc.
+   *
+   * <p>
+   */
+  class InnerInputCorrectJavaDocParagraphCheck {
+    // violation 3 lines above '<p> tag should be placed immediately before the first word'
+
+    // violation 4 lines below '<p> tag should be placed immediately before the first word'
+    /**
+     * Some Javadoc.
+     *
+     * <p>
+     *
+     * @since 8.0
+     */
+    public static final byte NUL = 0;
+
+    // violation 8 lines below '<p> tag should be placed immediately before the first word'
+    /**
+     * /^WARN/ Some Javadoc.
+     *
+     * <p>/^WARN/
+     *
+     * <p>/^WARN/ Some Javadoc.
+     *
+     * <p>
+     *
+     * @see <a
+     *     href="http://www.gwtproject.org/doc/latest/DevGuideOrganizingProjects.html#DevGuideModules">
+     *     Documentation about GWT emulated source</a>
+     */
+    boolean emulated() {
+      return false;
+    }
+  }
+
+  InnerInputCorrectJavaDocParagraphCheck anon =
+      new InnerInputCorrectJavaDocParagraphCheck() {
+
+        /**
+         * Some Javadoc.
+         *
+         * <p>Some Javadoc.
+         *
+         * @since 8.0
+         */
+        public static final byte NUL = 0;
+
+        // violation 4 lines below '<p> tag should be placed immediately before the first word'
+        /**
+         * /WARN/ Some Javadoc.
+         *
+         * <p>
+         *
+         * <p>Some Javadoc.
+         *
+         * @see <a
+         *     href="http://www.gwtproject.org/doc/latest/DevGuideOrganizingProjects.html#DevGuideModules">
+         *     Documentation about
+         *     <p>GWT emulated source</a>
+         */
+        // violation 2 lines above '<p> tag should be preceded with an empty line.'
```

Comment:
Is it a bug in the formatter?

Reply:
I think yes, rule [7.1.2 Paragraphs](https://google.github.io/styleguide/javaguide.html#s7.1.2-javadoc-paragraphs) says: 

> One blank line—that is, a line containing only the aligned leading asterisk (*)—appears between paragraphs, and before the group of block tags if present.

"group of block tags" can be @param @see @return, etc. So there should be a blank line between `<p>` and `@see`

good catch. Will open an issue regarding this on formatter's GH

Reply:
this is not valid javadoc.

```
rivanov@p5510:/var/tmp$ cat Test.java
public class Test {
    /**
    * The client's first name.
    *
    * @see <a
    *     href="http://www.gwtproject.org/doc/latest/DevGuideOrganizingProjects.html#DevGuideModules">
    *     Documentation about
    *     <p>GWT emulated source</a>
    */
  public String thirdName;
}
rivanov@p5510:/var/tmp$ 
rivanov@p5510:/var/tmp$ 
rivanov@p5510:/var/tmp$ javadoc Test.java
Loading source file Test.java...
Constructing Javadoc information...
Standard Doclet version 11.0.14.1
Building tree for all the packages and classes...
Generating ./Test.html...
Test.java:8: error: block element not allowed within inline element <a>: p
    *     <p>GWT emulated source</a>
          ^
Generating ./package-summary.html...
Generating ./package-tree.html...
Generating ./constant-values.html...
Building index for all the packages and classes...
Generating ./overview-tree.html...
Generating ./index-all.html...
Building index for all classes...
Generating ./allclasses-index.html...
Generating ./allpackages-index.html...
Generating ./deprecated-list.html...
Building index for all classes...
Generating ./allclasses.html...
Generating ./allclasses.html...
Generating ./index.html...
Generating ./help-doc.html...
1 error

```


so it is not a defect of formater.

Reply:
it is not a defect in Checkstyle as as have no validation for this at all for now. It should be some new Check, but let not create it now, let wait for users feedback.